src/main/java/com/amazon/opendistroforelasticsearch/sql/antlr/OpenDistroSqlAnalyzer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.antlr;
+
+import com.amazon.opendistroforelasticsearch.sql.antlr.parser.OpenDistroSqlLexer;
+import com.amazon.opendistroforelasticsearch.sql.antlr.parser.OpenDistroSqlParser;
+import com.amazon.opendistroforelasticsearch.sql.antlr.syntax.CaseInsensitiveCharStream;
+import com.amazon.opendistroforelasticsearch.sql.antlr.syntax.SyntaxAnalysisErrorListener;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.tree.ParseTree;
+
+/**
+ * Entry point for ANTLR generated parser to perform strict syntax and semantic analysis.
+ */
+public class OpenDistroSqlAnalyzer {
+
+    /**
+     * Generate parse tree for the query to perform syntax and semantic analysis.
+     * Runtime exception with clear message is thrown for any verification error.
+     *
+     * @param sql   original query
+     */
+    public void analyze(String sql) {
+        analyzeSemantic(
+            analyzeSyntax(
+                createParser(
+                    createLexer(sql))));
+    }
+
+    private OpenDistroSqlParser createParser(Lexer lexer) {
+        return new OpenDistroSqlParser(
+                   new CommonTokenStream(lexer));
+    }
+
+    private OpenDistroSqlLexer createLexer(String sql) {
+         return new OpenDistroSqlLexer(
+                    new CaseInsensitiveCharStream(sql));
+    }
+
+    private ParseTree analyzeSyntax(OpenDistroSqlParser parser) {
+        parser.addErrorListener(new SyntaxAnalysisErrorListener());
+        return parser.root();
+    }
+
+    private void analyzeSemantic(ParseTree tree) {
```

Comment:
I suggest adding a TODO: comment here and explaining why this method should be called, otherwise just remove it.

Reply:
> I suggest adding a TODO: comment here and explaining why this method should be called, otherwise just remove it.

Sure. Will add a TODO to clarify that this is a todo item for next stage.